app/controllers/ApiController.scala

```diff
@@ -26,7 +26,7 @@ class ApiController(
             val response = MatcherResponse(
               matches = matches,
               blocks = check.blocks,
-              categoryIds = check.categoryIds.getOrElse(matcherPool.getCurrentCategories.map { _._2.id })
+              categoryIds = matches.map(_.rule.category.id).toSet.toList
```

Comment:
I think `distinct` is more elegant here:
```suggestion
              categoryIds = matches.map(_.rule.category.id).distinct
```
I think the order of first instance is maintained, but since a set won't maintain any order I doubt you care.